eclipse-che/che-operator
# chore: Update DevWorkspace Operator dependency to v0.12.2

### What does this PR do?
* Update DWO to v0.12.2
* Update Che RoutingSolver to accomodate changes from https://github.com/devfile/devworkspace-operator/pull/706

### Screenshot/screencast of this PR
N/A


### What issues does this PR fix or reference?
Closes https://github.com/eclipse/che/issues/21039

### How to test this PR?
1. Deploy Che Operator with DevWorkspaces enabled
2. Create a Devfile 2.0 workspace
3. Stop the Devfile 2.0 workspace
4. Delete the workspace
5. Verify that DevWorkspace resource on cluster is removed, and that finalizer on `DevWorkspaceRouting` is cleared

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
